### PR TITLE
Containerized model interface and sample container model

### PR DESCRIPTION
**Pull request by benlipkin**
Hi Team,

Here is a preliminary end-to-end interface supporting containerized model inference (and a sample container model). I've mostly taken the existing code from the `HuggingFaceSubject` model helper class and modified it to run a subprocess on a container rather than parsing the outputs of an HF CausalLM. I've added some preliminary docstrings describing the requirements of the interface, and a set of test cases that mirror what is currently being evaluated for HF models. I have tried adding singularity to the travis file, so we can see if those tests run, or if additional environment management is required. 

I also want to note that this is not an optimal implementation in terms of runtime efficiency, but is most consistent with how code is currently being structured in the brainscore model helper classes. I have several design patterns in mind to accelerate this considerably, but they are currently out of scope as I need to focus on building out the RNNG model for which this container interface was primarily developed at this stage. The goal of this PR is a complete end-to-end implementation supporting containerized model inference on our core tasks, so that we can focus on model-side features for the time being.

Let me know how this looks, and I look forward to using this interface to integrate exciting new models, as well as optimizing runtime efficiency in coming months.

### CODE REVIEW

File: tests/test_integration.py [hunk 1]
```diff
@@ -17,8 +17,8 @@
 def test_score(model_identifier, benchmark_identifier, expected_score):
     actual_score = score(model_identifier=model_identifier, benchmark_identifier=benchmark_identifier)
     assert actual_score == expected_score
-
-
+    
+
```
Comment (mschrimpf) [inline tests/test_integration.py:21]: this looks like an unnecessary change 😛

File: brainscore_language/model_helpers/container.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    Task: extract representation
+    Input: --measure MEASURE
+    Output: {"measure": REPRESENTATION}
+    where MEASURE is the name of the representation as supported by your container 
+    and REPRESENTATION is an array of shape (1, representation_size) cast to a list
+
+    Note: While the internals of any containerized model are not restricted, the interface must be as described above. 
+    It is highly recommended to raise detailed error messages from inside the container, so they can be escalated here.
+    It is also recommended to include a list of supported measures in the container's documentation.
+    """
```
Comment (mschrimpf) [inline brainscore_language/model_helpers/container.py:58]: love it. Maybe you could add a small blurb on how to add new measures, e.g. when we define a new task (a near-term one is likely going to be next-word surprisal)

File: brainscore_language/model_helpers/container.py [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+            cmd = ["docker", "pull", f"{self._container}"]
+        elif self._backend == "singularity":
+            f = self._get_singularity_container(self._cachedir, self._container)
+            if f.exists():
+                self._logger.info(f"Container already downloaded to {self._cachedir}.")
+                return
+            cmd = ["singularity", "pull", f"docker://{self._container}"]
+        else:
+            raise RuntimeError(f"Unknown container backend {self._backend}")
+        try:
```
Comment (mschrimpf) [inline brainscore_language/model_helpers/container.py:145]:
```suggestion

        # run command
        try:
```

**Review by mschrimpf — APPROVED** [review]
Fantastic. Only have a few minor suggestions, but looks good to go from my side.

File: tests/test_integration.py (hunk 1 above)
Comment (benlipkin) [reply tests/test_integration.py:21]: fixed thanks.

File: brainscore_language/model_helpers/container.py (hunk 2 above)
Comment (benlipkin) [reply brainscore_language/model_helpers/container.py:58]: added a paragraph will commit once I finish these comments.

File: brainscore_language/models/rnnlm/__init__.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from brainscore_language import model_registry, ArtificialSubject
+from brainscore_language.model_helpers.container import ContainerSubject
+
+
+model_registry["rnn-lm-ptb"] = lambda: ContainerSubject(
+    container="benlipkin/rnng:6f6825d1c4a8c58c844b4b82123b967bb0bab6ce",
```
[the remark this answers is not in the document]
Comment (benlipkin) [reply brainscore_language/models/rnnlm/__init__.py:6]: yep, the hash matches the git commit that triggered the container build. an update to the model repo would not overwrite an older version

**Comment by benlipkin** [comment]
Thanks Martin! all updated, will merge once tests confirmed passing on new commits.
